editor/plugins/style_box_editor_plugin.cpp

```diff
@@ -81,41 +70,46 @@ void StyleBoxPreview::_notification(int p_what) {
 				// See https://github.com/godotengine/godot/issues/50743.
 				break;
 			}
-			grid_preview->set_texture_normal(get_theme_icon(SNAME("StyleBoxGridInvisible"), SNAME("EditorIcons")));
-			grid_preview->set_texture_pressed(get_theme_icon(SNAME("StyleBoxGridVisible"), SNAME("EditorIcons")));
-			grid_preview->set_texture_hover(get_theme_icon(SNAME("StyleBoxGridVisible"), SNAME("EditorIcons")));
-			checkerboard->set_texture(get_theme_icon(SNAME("Checkerboard"), SNAME("EditorIcons")));
+			set_texture(get_theme_icon(SNAME("Checkerboard"), SNAME("EditorIcons")));
+			grid_preview->set_icon(get_theme_icon(SNAME("StyleBoxGrid"), SNAME("EditorIcons")));
+		} break;
+		case NOTIFICATION_DRAW: {
+			_redraw();
 		} break;
 	}
 }
 
 void StyleBoxPreview::_redraw() {
 	if (stylebox.is_valid()) {
-		Ref<Texture2D> grid_texture_disabled = get_theme_icon(SNAME("StyleBoxGridInvisible"), SNAME("EditorIcons"));
-		Rect2 preview_rect = preview->get_rect();
-		preview_rect.position += grid_texture_disabled->get_size();
-		preview_rect.size -= grid_texture_disabled->get_size() * 2;
+		float grid_button_width = get_theme_icon(SNAME("StyleBoxGrid"), SNAME("EditorIcons"))->get_size().x;
+		Rect2 preview_rect = get_rect();
+		preview_rect = preview_rect.grow(-grid_button_width);
 
-		// Re-adjust preview panel to fit all drawn content
-		Rect2 draw_rect = stylebox->get_draw_rect(preview_rect);
-		preview_rect.size -= draw_rect.size - preview_rect.size;
-		preview_rect.position -= draw_rect.position - preview_rect.position;
+		// Re-adjust preview panel to fit all drawn content.
+		Rect2 drawing_rect = stylebox->get_draw_rect(preview_rect);
+		preview_rect.size -= drawing_rect.size - preview_rect.size;
+		preview_rect.position -= drawing_rect.position - preview_rect.position;
 
-		preview->draw_style_box(stylebox, preview_rect);
+		draw_style_box(stylebox, preview_rect);
 
 		Ref<StyleBoxTexture> sbt = stylebox;
 		if (sbt.is_valid() && grid_preview->is_pressed()) {
```

Comment:
I think to really make it more readable it would be good to add comments in there, describing what it is that we are rendering in each block of methods.